### PR TITLE
#3 Add DocFX and C4-Builder docs

### LINKED ISSUE #3: Add DocFX and C4-Builder docs
**As a** developer,
**I want** publicly available documentation created with DocFX and C4-Builder that is automatically built and deployed on GitHub Pages
**so that** I have easy access to documentation for maintenance, reading and further development.

### PR DESCRIPTION
**Pull request by krzysztofzajaczkowski**
Additionally, added basic structure, diagram and diagram description to better show off template and possibilities for further documentation.

closes #3

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        
+    - name: Build Documentation
+      uses: nikeee/docfx-action@v1.0.0
+      with:
+        args: docs/docfx.json
+      continue-on-error: false
+
+    - name: Publish
+      uses: peaceiris/actions-gh-pages@v3
+      # if: ${{ github.event.pull_request.merged }}
```
Comment (krzysztofzajaczkowski) [inline .github/workflows/publish-docs.yml]: Uncomment before merging and after ensuring workflow success.

File: .github/workflows/publish-docs.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Publish Docs on GitHub Pages
+
+on:
+  pull_request:
```
Comment (RafKulas) [inline .github/workflows/publish-docs.yml:4]: TODO

File: docs/docfx.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        {
+		  "src": "../",
+          "files": [
+            "src/**.csproj"
+          ],
+		  "exclude": [
+		    "src/**Tests*.csproj"
+		  ]
+        }
+      ],
```
Comment (RafKulas) [inline docs/docfx.json]: Fix indentation
image

**Review by RafKulas — APPROVED** [review]

File: docs/docfx.json (hunk 3 above)
Comment (krzysztofzajaczkowski) [reply docs/docfx.json]: Fixed

File: .github/workflows/publish-docs.yml (hunk 2 above)
Comment (krzysztofzajaczkowski) [reply .github/workflows/publish-docs.yml:4]: I think it's better to trigger building flow on pull requests and deploy new docs version only on merge.

**Review by RafKulas — APPROVED** [review]

**Pull request merged**